mesonbuild/compilers/asm.py

```diff
@@ -0,0 +1,83 @@
+import typing as T
+
+from ..mesonlib import EnvironmentException, MachineChoice
+from .compilers import Compiler
+
+if T.TYPE_CHECKING:
+    from ..coredata import KeyedOptionDictType
+    from ..environment import Environment
+
+
+class NasmCompiler(Compiler):
+    language = 'asm'
+    id = 'nasm'
+
+    def needs_static_linker(self) -> bool:
+        return False
+
+    def get_always_args(self) -> T.List[str]:
+        if self.info.is_windows():
+            plat = 'win'
+        elif self.info.is_linux() or self.info.is_android():
+            plat = 'elf'
+        elif self.info.is_darwin():
+            plat = 'macho'
+        else:
+            raise EnvironmentException(f'Unsupported system {self.info.system}')
+        cpu = '64' if self.info.is_64_bit else '32'
+        return [f'-f{plat}{cpu}']
+
+    def get_werror_args(self) -> T.List[str]:
+        return ['-Werror']
+
+    def get_output_args(self, outputname: str) -> T.List[str]:
+        return ['-o', outputname]
+
+    def get_optimization_args(self, optimization_level: str) -> T.List[str]:
+        return [f'-O{optimization_level}']
+
+    def get_depfile_suffix(self) -> str:
+        return 'd'
+
+    def get_dependency_gen_args(self, outtarget: str, outfile: str) -> T.List[str]:
+        return ['-MD', '-MQ', outtarget, '-MF', outfile]
+
+    def sanity_check(self, work_dir: str, environment: 'Environment') -> None:
```

Comment:
since nasm only works on x86 (I think), it might be worth short cirucuiting here since we have the environment and can check if we're on x86 or x86_64

Reply:
Good point. That code was just place holder I copied form nasm tutorial to see if it works.

Reply:
done.